### PR TITLE
doc about filter performance

### PR DESCRIPTION
**Pull request by kazchimo**
## Description
`Arbitrary.filter` can harm the performance of generations of values, and this is shown in [the advanced doc](https://github.com/dubzzz/fast-check/blob/master/documentation/AdvancedArbitraries.md#filter-values) but not in the code doc.
This is a very important and fundamental issue when use property based testing libraries so should be written in the code doc explicitly.
The doc content is borrowed from the advanced doc description.

## In a nutshell

❌ New feature
❌ Fix an issue
✔️ Documentation improvement
❌ Other: *please explain*

### CODE REVIEW

**Review by dubzzz — APPROVED** [review]

**Comment by dubzzz** [comment]
Thanks a lot for the improvement!
